app/grandchallenge/core/utils/access_requests.py

```diff
@@ -40,10 +40,13 @@ def process_access_request(request_object):
             if request_object.user.verification.is_verified:
                 # immediately allow access, no need for a notification
                 request_object.status = request_object.ACCEPTED
-                request_object.save()
-                return
+            else:
+                request_object.status = request_object.REJECTED
+            request_object.save()
+            return
         except ObjectDoesNotExist:
-            pass
+            request_object.status = request_object.REJECTED
+            request_object.save()

```

Comment:
They should not be automatically rejected. People cannot make a request again in this case.